bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/BundleVersion.java

```diff
@@ -0,0 +1,147 @@
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.addon.marketplace;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link BundleVersion} wraps a bundle version and provides a method to compare them
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public class BundleVersion {
+    private static final Pattern VERSION_PATTERN = Pattern.compile(
+            "(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<micro>\\d+)(\\.((?<rc>RC)|(?<milestone>M))?(?<qualifier>\\d+))?");
+    public static final Pattern RANGE_PATTERN = Pattern.compile(
+            "\\[(?<start>\\d+\\.\\d+(?<startmicro>\\.\\d+(\\.\\w+)?)?);(?<end>\\d+\\.\\d+(?<endmicro>\\.\\d+(\\.\\w+)?)?)(?<endtype>[)\\]])");
+
+    private final Logger logger = LoggerFactory.getLogger(BundleVersion.class);
+
+    private final int major;
+    private final int minor;
+    private final int micro;
+    private final @Nullable Long qualifier;
+
+    public BundleVersion(String version) {
+        Matcher matcher = VERSION_PATTERN.matcher(version);
+        if (matcher.matches()) {
+            this.major = Integer.parseInt(matcher.group("major"));
+            this.minor = Integer.parseInt(matcher.group("minor"));
+            this.micro = Integer.parseInt(matcher.group("micro"));
+            String qualifier = matcher.group("qualifier");
+            if (qualifier != null) {
+                long intQualifier = Long.parseLong(qualifier);
+                if (matcher.group("rc") != null) {
+                    // we can safely assume that there are less than Integer.MAX_VALUE milestones
+                    // so RCs are always newer than milestones
+                    // since snapshot qualifiers are larger than 10*Integer.MAX_VALUE they are
+                    // still considered newer
+                    this.qualifier = intQualifier + Integer.MAX_VALUE;
+                } else {
+                    this.qualifier = intQualifier;
+                }
+            } else {
+                this.qualifier = null;
+            }
+        } else {
+            throw new IllegalArgumentException("Input does not match pattern");
+        }
+    }
+
+    public boolean inRange(@Nullable String range) {
+        if (range == null || range.isBlank()) {
+            // if no range is given, we assume the range covers everything
+            return true;
+        }
+        Matcher matcher = RANGE_PATTERN.matcher(range);
+        if (!matcher.matches()) {
+            logger.warn("Argument {} does not define a valid version range. Assuming inRange", range);
```

Comment:
If you'd throw a `IllegalArgumentException` here, you could catch it and add a bit more context to the logging so it is easier to troubleshoot which add-on has an invalid range.